framework/src/utils/SolutionInvalidityRegistry.C

```diff
@@ -0,0 +1,41 @@
+#include "SolutionInvalidityRegistry.h"
+
+#include "DataIO.h"
+
+namespace moose
+{
+namespace internal
+{
+
+SolutionInvalidityRegistry &
+getSolutionInvalidityRegistry()
+{
+  // In C++11 this is even thread safe! (Lookup "Static Initializers")
+  static SolutionInvalidityRegistry solution_invalid_registry_singleton;
+
+  return solution_invalid_registry_singleton;
+}
+
+SolutionInvalidityRegistry::SolutionInvalidityRegistry()
+  : GeneralRegistry<std::string, SolutionInvalidityInfo>("SolutionInvalidityRegistry")
+{
+}
+
+InvalidSolutionID
+SolutionInvalidityRegistry::registerInvalidObject(const std::string & object_name,
+                                                  const std::string & message)
+{
+  return actuallyRegisterSection(object_name, message);
+}
+
+InvalidSolutionID
+SolutionInvalidityRegistry::actuallyRegisterSection(const std::string & object_name,
+                                                    const std::string & message)
+{
+  const auto create_item = [&object_name, &message](const std::size_t id)
+  { return SolutionInvalidityInfo(id, object_name, message); };
+  return registerItem(object_name, create_item);
+}
```

Comment:
Let's get rid of `actuallyRegisterSection` and move all of its code into `registerInvalidObject`, and then rename `registerInvalidObject` into `registerInvalidity`